src/tlp_lib/smartcontracts/EthereumSC.py

```diff
@@ -128,12 +122,11 @@ def add_solution(self, solution: GCTLP_Encrypted_Message, witness: TLP_Digest) -
         if not self._has_succeeded(self._contract.functions.addSolution(solution, witness)):
             raise RuntimeError("Solution was not added correctly")
 
-    def get_message_at(self, i: int) -> GCTLP_Encrypted_Message:
-        return self._contract.functions.getSolutionAt(i).call()[0]
+    def verify_solution(self, i: int, /) -> bool:
+        return self._has_succeeded(self._contract.functions.verifySolution(i))
 
-    def pay(self, i: int) -> None:
-        if not self._has_succeeded(self._contract.functions.pay(i)):
-            raise RuntimeError("Payout was not successful")
+    def get_message_at(self, i: int) -> GCTLP_Encrypted_Message:
+        return self._contract.functions.getSolutionAt(i).call()
 
     def pay_back(self, i: int) -> None:
```

Comment:
ditto on `pay_back` going away to be replaced with `pay` that pays the appropriate party